core/views.py

```diff
@@ -446,6 +448,44 @@ def get_object(self):
 
 profile_edit = ProfileEditView.as_view()
 
+class ConnectedUserCircleEditView(UpdateView):
+    model = models.User  # or CircleMembership, depending on your data model
+    form_class = forms.ConnectedUserCircleForm
+    template_name = 'core/edit_connected_user_circles.html'
+    success_url = reverse_lazy('connection_list')  # Adjust as necessary
+
+    def get_object(self):
+        # Ensure the user being edited is connected to the request.user
+        target_user = get_object_or_404(models.User, pk=self.kwargs['pk'])
+        if not self.request.user.is_connected_with(target_user):
+            raise PermissionDenied
```

Comment:
As a general rule, if a user doesn't have permissions to view an object, attempts to operate on that object should return 404 Not found, because you don't even want to confirm that the object exists. We'd only be leaking the pk in this case, so it's not a big deal, but one other thing to note is that `is_connected_with` emits another DB query. If we do this, it checks both existence and permissions in 1 query:

```
def get_object(self):
    return get_object_or_404(
        self.request.user.connected_users,
        pk=self.kwarg['pk'],
    )
```

Permission Denied is more for situations like, the user is trying to view a page that requires them to be logged in.